callback/api_nf_subscribe_notify.go

```diff
@@ -0,0 +1,60 @@
+package callback
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/omec-project/http_wrapper"
+	"github.com/omec-project/openapi"
+	"github.com/omec-project/openapi/models"
+	"github.com/omec-project/smf/logger"
+	"github.com/omec-project/smf/producer"
+)
+
+func HTTPNfSubscriptionStatusNotify(c *gin.Context) {
+	var nfSubscriptionStatusNotification models.NotificationData
+
+	requestBody, err := c.GetRawData()
+	if err != nil {
+		logger.PduSessLog.Errorf("Get Request Body error: %+v", err)
+		problemDetail := models.ProblemDetails{
+			Title:  "System failure",
+			Status: http.StatusInternalServerError,
+			Detail: err.Error(),
+			Cause:  "SYSTEM_FAILURE",
+		}
+		c.JSON(http.StatusInternalServerError, problemDetail)
+		return
+	}
+
+	err = openapi.Deserialize(&nfSubscriptionStatusNotification, requestBody, "application/json")
+	if err != nil {
+		problemDetail := "[Request Body] " + err.Error()
+		rsp := models.ProblemDetails{
+			Title:  "Malformed request syntax",
+			Status: http.StatusBadRequest,
+			Detail: problemDetail,
+		}
+		logger.PduSessLog.Errorln(problemDetail)
+		c.JSON(http.StatusBadRequest, rsp)
+		return
+	}
+
+	req := http_wrapper.NewRequest(c.Request, nfSubscriptionStatusNotification)
+
+	rsp := producer.HandleNfSubscriptionStatusNotify(req)
+
+	responseBody, err := openapi.Serialize(rsp.Body, "application/json")
+	if err != nil {
+		logger.PduSessLog.Errorln(err)
```

Comment:
Error log with some content shall be helpful

Reply:
Added